src/components/Task/TaskList/TaskList.tsx

```diff
@@ -0,0 +1,91 @@
+import "./TaskList.css";
+
+import { useState, useRef } from "react";
+import { CheckListCircle } from "../../../assets/checkListCircle";
+import EditIcon from "../../../assets/Edit-Icon";
+import DeleteButton from "../../../assets/Delete-Icon";
+
+interface TaskListProps {
+  index: number;
+  task: { id: number; name: string; completed: boolean };
+  onToggleTask: (index: number) => void;
+  onDeleteTask: (index: number) => void;
+  onEditTask: (index: number, newName: string) => void;
+}
+
+function TaskList({
+  index,
+  task,
+  onToggleTask,
+  onDeleteTask,
+  onEditTask,
+}: TaskListProps) {
+  const [isEditing, setIsEditing] = useState(false);
+  const taskNameRef = useRef<HTMLParagraphElement>(null);
+
+  const handleEdit = () => {
+    console.log("Edit button clicked");
+    setIsEditing(true);
+    setTimeout(() => {
+      taskNameRef.current?.focus();
+    }, 0);
+  };
+
+  const handleEditSubmit = () => {
+    if (taskNameRef.current) {
+      const newTaskName = taskNameRef.current.textContent || task.name;
+      if (newTaskName.length < 3) {
+        alert("Task name must be at least 3 characters long");
```

Comment:
why are you not handling this yup validations?

Reply:
I handled the add task component with yup validation . I knew this could be done using yup too but this method seemed more efficient to me. I will do it with yup then

Reply:
@faizanmunir017 we should maintain code consistency everywhere, if you are using yup for form validations and error handling please stick to it everywhere.